sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -708,6 +736,26 @@ case class HashShuffleSpec(
   override def numPartitions: Int = partitioning.numPartitions
 }
 
+case class CoalescedHashShuffleSpec(
+    from: ShuffleSpec,
+    @transient partitions: Seq[CoalescedBoundary]) extends ShuffleSpec {
```

Comment:
what's the rationale of `@transient` here?

Reply:
I added because of the suggestion of @ulysses-you here https://github.com/apache/spark/pull/43435#discussion_r1368494061

Reply:
It is to avoid unnecessary ser/de. We only use it to check if compatible with other shuffle spec at driver side.

Reply:
which plan may hold `ShuffleSpec`? and do we need to do the same for other `ShuffleSpec`s?

Reply:
I thought it would be holden at some place... After some search, it seems ShuffleSpec is created dynamically so it's unnecessary to consider ser/de. We can remove it @transient.